app/Community/Concerns/HasViews.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Community\Concerns;
+
+use App\Models\User;
+use App\Models\Viewable;
+use Illuminate\Database\Eloquent\Relations\MorphMany;
+
+/**
+ * Provides polymorphic view tracking for models.
+ *
+ * This trait allows models to track when individual users have viewed them,
+ * storing timestamps in the `viewables` table. Useful for features like
+ * unread indicators, view counts, or determining if a user has seen content.
+ *
+ * @example
+ * ```php
+ * class News extends BaseModel {
+ *     use HasViews;
+ * }
+ *
+ * $news->markAsViewedBy($user);
+ * if ($news->wasViewedBy($user)) {
+ *     // ... user has viewed this news item ...
+ * }
+ * ```
+ */
+trait HasViews
+{
+    /**
+     * @return MorphMany<Viewable, $this>
+     */
+    public function views(): MorphMany
+    {
+        return $this->morphMany(Viewable::class, 'viewable');
+    }
+
+    /**
+     * Mark this item as viewed by the given user.
+     */
+    public function markAsViewedBy(User $user): void
+    {
+        $this->views()->firstOrCreate(
+            [
+                'user_id' => $user->id,
+            ],
+            [
+                'viewed_at' => now(),
+            ]
+        );
```

Comment:
For news items (especially the site updates), can we just track the most recently viewed news item? Either by updating the viewable_id in the row, or deleting the older rows?

We get roughly 1 news post per day. If 10,000 Users hit the home page every day, that's going to be ~10,000 viewable records created per day, or 300,000 per month just for the news posts.

Reply:
Updated the trait in latest to support this behavior. In the _News.php_ model file, we now have:
```php
public function shouldTrackLatestViewOnly(): bool
{
    return $this->category === NewsCategory::SiteReleaseNotes;
}
```

This mode defaults to `false`. When enabled, on tracking a view, all previous rows for the user+entity will be deleted.